shell/platform/android/android_external_view_embedder.cc

```diff
@@ -0,0 +1,84 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/platform/android/android_external_view_embedder.h"
+
+#include "fml/trace_event.h"
+
+namespace flutter {
+
+// |ExternalViewEmbedder|
+void AndroidExternalViewEmbedder::PrerollCompositeEmbeddedView(
+    int view_id,
+    std::unique_ptr<EmbeddedViewParams> params) {
+  TRACE_EVENT0("flutter",
+               "AndroidExternalViewEmbedder::PrerollCompositeEmbeddedView");
+
+  picture_recorders_[view_id] = std::make_unique<SkPictureRecorder>();
+  picture_recorders_[view_id]->beginRecording(SkRect::Make(frame_size_));
+
+  composition_order_.push_back(view_id);
+}
+
+// |ExternalViewEmbedder|
+SkCanvas* AndroidExternalViewEmbedder::CompositeEmbeddedView(int view_id) {
+  return picture_recorders_[view_id]->getRecordingCanvas();
+}
+
+// |ExternalViewEmbedder|
+std::vector<SkCanvas*> AndroidExternalViewEmbedder::GetCurrentCanvases() {
+  std::vector<SkCanvas*> canvases;
+  for (size_t i = 0; i < composition_order_.size(); i++) {
+    int64_t view_id = composition_order_[i];
+    canvases.push_back(picture_recorders_[view_id]->getRecordingCanvas());
+  }
+  return canvases;
+}
+
+// |ExternalViewEmbedder|
+bool AndroidExternalViewEmbedder::SubmitFrame(GrContext* context,
+                                              SkCanvas* background_canvas) {
+  TRACE_EVENT0("flutter", "AndroidExternalViewEmbedder::SubmitFrame");
+  for (size_t i = 0; i < composition_order_.size(); i++) {
+    int64_t view_id = composition_order_[i];
+    background_canvas->drawPicture(
+        picture_recorders_[view_id]->finishRecordingAsPicture());
+  }
+  return true;
+}
+
+// |ExternalViewEmbedder|
+PostPrerollResult AndroidExternalViewEmbedder::PostPrerollAction(
+    fml::RefPtr<fml::RasterThreadMerger> raster_thread_merger) {
+  return PostPrerollResult::kSuccess;
+}
+
+// |ExternalViewEmbedder|
+SkCanvas* AndroidExternalViewEmbedder::GetRootCanvas() {
+  // On Android, the root surface is created from the on-screen render target.
+  return nullptr;
+}
+
+// |ExternalViewEmbedder|
+void AndroidExternalViewEmbedder::BeginFrame(SkISize frame_size,
+                                             GrContext* context,
+                                             double device_pixel_ratio) {
+  frame_size_ = frame_size;
+}
+
+// |ExternalViewEmbedder|
+void AndroidExternalViewEmbedder::CancelFrame() {
+  composition_order_.clear();
+}
+
+// |ExternalViewEmbedder|
+void AndroidExternalViewEmbedder::FinishFrame() {
+  composition_order_.clear();
```

Comment:
Clear the frame size and picture recorders here as well. You could probably dry up this code by moving all the resetting stuff to ClearFrame and have FinishFrame call clear frame before doing anything else (which I suppose is nothing for now).

Reply:
👍I missed some state for sure last night